### PR TITLE
add limit for write to leddevice

### PR DESCRIPTION
**Pull request by redPanther**
**1.** Tell us something about your changes.
... writing to fast to a led device can led to problems.
For arduino/adalight is 20ms a good value. (my experience - otherwise I got flashes)

additional:
- extend xmas effect
- rework some effects to adapt to device latchTime
- fading for random.py

**2.** If this changes affect the .conf file. Please provide the changed section
yes, new option "latchTime" to "device" section

**3.** Reference an issue (optional)

Note: For further discussions use our forum: forum.hyperion-project.org

### CODE REVIEW

File: effects/schema/x-mas.schema.json [hunk 1]
```diff
@@ -5,11 +5,47 @@
...
+			"format":"colorpicker",
+			"default": [255,0,0],
+			"items" : {
+				"type": "integer",
+				"minimum": 0,
+				"maximum": 255
+			},
+			"minItems": 3,
+			"maxItems": 3,
+			"propertyOrder" : 3
```
Comment (brindosch) [inline effects/schema/x-mas.schema.json]: propertyOrder of line 34

**Comment by penfold42** [comment]
Is this the same as latchtime but done globally ?

Maybe we could combine them.

The time needed for store and forward devices like an arduino will need to tuned based on the number of leds

**Comment by redPanther** [comment]
Its the same. I uèdate the code later

**Comment by redPanther** [comment]
I want to remove latchtime. Do we need ns precision?
Btw the diff between latchtime and this implemention is, that latchtime blocks the whole hyperion, while the other just drops the frame.

**Comment by penfold42** [comment]
Nah...  

Latchtimes from datasheets I had handy: 
Ws2801 - 500 uSec
Ws2812 - 50 uSec
Ws2813 - 300 uSec
Apa102 - none - it's sent as end frame data bytes

**Comment by redPanther** [comment]
setting 1ms or perhaps better 5ms will satisfy all needs. To be flexible I will change the code, so every device can have its own default value for latchtime.

**Comment by penfold42** [comment]
Even at 5ms we should be ok. At some point it will be a limiting factor on the frame rate.

200 leds * 24 bits = 4800 bits
At 1,000,000 bit/second = 4.8 ms
That's 208 fps 

Adding 5ms of default wait time is still > 100 fps

**Comment by redPanther** [comment]
now old latchTime is gone and new latchTime is implemented. For the device specifc latchTime params (min, max, default) I need suggestions. Currently I guessed for most devices.

**Comment by penfold42** [comment]
Does the latchtimer start after the write is started or completed ?

**Comment by redPanther** [comment]
It starts before write. The check is also before write. If timeout is not reached, the device write is not called.
So the latchtime includes the data transfer.

**Pull request merged**